src/providers/auth/auth.provider.jsx

```diff
@@ -0,0 +1,17 @@
+import React, { useReducer } from 'react';
+import authReducer, { InitialState } from './auth.reducer';
+
+const AuthContext = React.createContext(null);
+
+function AuthProvider({ children }) {
+  const [authState, authDispatch] = useReducer(authReducer, InitialState);
```

Comment:
You have defined the actions, the types, the reducers and the context in their own file, this makes the code easy to maintain, good one